metadata/metadata.go

```diff
@@ -824,14 +858,49 @@ func (this *modelResource) Notify(lookup ResourceLookup, op operation, that Reso
 }
 
 func (resource *modelResource) UpdateStatus(status pb.ResourceStatus) error {
-	resource.serialized.Status = &status
 	return nil
 }
 
 func (resource *modelResource) UpdateSchedule(schedule string) error {
 	return fmt.Errorf("not implemented")
 }
 
+func (resource *modelResource) Update(lookup ResourceLookup, updateRes Resource) error {
+	deserialized := updateRes.Proto()
+	updateModel, ok := deserialized.(*pb.Model)
+	if !ok {
+		return errors.New("failed to deserialize existing model record")
+	}
+	updated := unionNameVariants(resource.serialized.Features, updateModel.Features)
```

Comment:
you can combine this and the line below if you'd like, keeps you from re-using updated